src/actions/request.rs

```diff
@@ -215,9 +215,33 @@ impl Runnable for Request {
         });
 
         if let Some(&SetCookie(ref cookies)) = response.headers.get::<SetCookie>() {
-          if let Some(cookie) = cookies.iter().next() {
-            let value = String::from(cookie.split(";").next().unwrap());
-            context.insert("cookie".to_string(), Yaml::String(value));
+          let context_cookies = context.entry("cookies".to_string()).or_insert_with(|| Yaml::Array(Vec::new()));
+          if let Yaml::Array(array) = context_cookies {
+            let mut existing = array
+              .iter()
+              .enumerate()
+              .map(|(index, cookie)| {
+                if let Yaml::String(cookie) = cookie {
+                  let key = cookie.split('=').next().unwrap().trim().to_string();
+                  (key, index)
+                } else {
+                  panic!("Unexpected cookie element {:?}", cookie);
+                }
+              })
+              .collect::<HashMap<String, usize>>();
+            for cookie in cookies.iter() {
+              for cook in cookie.split(';') {
+                let key = cook.split('=').next().unwrap().trim();
+                if let Some(index) = existing.get(key) {
+                  array[*index] = Yaml::String(cook.to_string());
+                } else {
+                  existing.insert(key.to_string(), array.len());
+                  array.push(Yaml::String(cook.to_string()));
+                }
+              }
+            }
+          } else {
+            panic!("The cookies context must be an array");
```

Comment:
We need to tweak some plans in the examples folder for this breaking change.

Reply:
The change is transparent for the cookie example. What would you like to see?

Reply:
Ignore me. I thought there was an example setting cookies from test plan, but is not there 😅

Reply:
If you want kind of backward compatibility, it is possible to transform an old `Yaml::String` into a `Yaml::Hash`, but that won't fix the case where there is already a hash. This case is unlikely though.

Reply:
It looks much better 💯